cmake/kokkos_arch.cmake

```diff
@@ -330,13 +330,41 @@ CHECK_CUDA_ARCH(PASCAL61  sm_61)
 CHECK_CUDA_ARCH(VOLTA70   sm_70)
 CHECK_CUDA_ARCH(VOLTA72   sm_72)
 CHECK_CUDA_ARCH(TURING75  sm_75)
+
+SET(AMDGPU_ARCH_ALREADY_SPECIFIED "")
+FUNCTION(CHECK_AMDGPU_ARCH ARCH FLAG)
+  IF(KOKKOS_ARCH_${ARCH})
+    IF(AMDGPU_ARCH_ALREADY_SPECIFIED)
+      MESSAGE(FATAL_ERROR "Multiple GPU architectures given! Already have ${AMDGPU_ARCH_ALREADY_SPECIFIED}, but trying to add ${ARCH}. If you are re-running CMake, try clearing the cache and running again.")
+    ENDIF()
+    SET(AMDGPU_ARCH_ALREADY_SPECIFIED ${ARCH} PARENT_SCOPE)
+    IF (NOT KOKKOS_ENABLE_HIP AND NOT KOKKOS_ENABLE_OPENMPTARGET)
+      MESSAGE(WARNING "Given HIP arch ${ARCH}, but Kokkos_ENABLE_AMDGPU and Kokkos_ENABLE_OPENMPTARGET are OFF. Option will be ignored.")
+      UNSET(KOKKOS_ARCH_${ARCH} PARENT_SCOPE)
+    ELSE()
+      SET(KOKKOS_AMDGPU_ARCH_FLAG ${FLAG} PARENT_SCOPE)
+      GLOBAL_APPEND(KOKKOS_AMDGPU_OPTIONS "${AMDGPU_ARCH_FLAG}=${FLAG}")
+    ENDIF()
+  ENDIF()
+ENDFUNCTION()
```

Comment:
```diff
--- check_cuda_arch
+++ check_amdgpu_arch
@@ -1,19 +1,16 @@
-SET(CUDA_ARCH_ALREADY_SPECIFIED "")
-FUNCTION(CHECK_CUDA_ARCH ARCH FLAG)
+SET(AMDGPU_ARCH_ALREADY_SPECIFIED "")
+FUNCTION(CHECK_AMDGPU_ARCH ARCH FLAG)
   IF(KOKKOS_ARCH_${ARCH})
-    IF(CUDA_ARCH_ALREADY_SPECIFIED)
-      MESSAGE(FATAL_ERROR "Multiple GPU architectures given! Already have ${CUDA_ARCH_ALREADY_SPECIFIED}, but trying to add ${ARCH}. If you are re-running CMake, try clearing the cache and running again.")
+    IF(AMDGPU_ARCH_ALREADY_SPECIFIED)
+      MESSAGE(FATAL_ERROR "Multiple GPU architectures given! Already have ${AMDGPU_ARCH_ALREADY_SPECIFIED}, but trying to add ${ARCH}. If you are re-running CMake, try clearing the cache and running again.")
     ENDIF()
-    SET(CUDA_ARCH_ALREADY_SPECIFIED ${ARCH} PARENT_SCOPE)
-    IF (NOT KOKKOS_ENABLE_CUDA AND NOT KOKKOS_ENABLE_OPENMPTARGET)
-      MESSAGE(WARNING "Given CUDA arch ${ARCH}, but Kokkos_ENABLE_CUDA and Kokkos_ENABLE_OPENMPTARGET are OFF. Option will be ignored.")
+    SET(AMDGPU_ARCH_ALREADY_SPECIFIED ${ARCH} PARENT_SCOPE)
+    IF (NOT KOKKOS_ENABLE_HIP AND NOT KOKKOS_ENABLE_OPENMPTARGET)
+      MESSAGE(WARNING "Given HIP arch ${ARCH}, but Kokkos_ENABLE_AMDGPU and Kokkos_ENABLE_OPENMPTARGET are OFF. Option will be ignored.")
       UNSET(KOKKOS_ARCH_${ARCH} PARENT_SCOPE)
     ELSE()
-      SET(KOKKOS_CUDA_ARCH_FLAG ${FLAG} PARENT_SCOPE)
-      GLOBAL_APPEND(KOKKOS_CUDA_OPTIONS "${CUDA_ARCH_FLAG}=${FLAG}")
-      IF(KOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE)
-        GLOBAL_APPEND(KOKKOS_LINK_OPTIONS "${CUDA_ARCH_FLAG}=${FLAG}")
-      ENDIF()
+      SET(KOKKOS_AMDGPU_ARCH_FLAG ${FLAG} PARENT_SCOPE)
+      GLOBAL_APPEND(KOKKOS_AMDGPU_OPTIONS "${AMDGPU_ARCH_FLAG}=${FLAG}")
     ENDIF()
   ENDIF()
 ENDFUNCTION()
```

Maybe add a comment that about omitting RDC because we might need to add that to the HIP backend.

Reply:
They don't support that yet do they? I think we will remember to fix this when we add support for relocatable device code to HIP.